### DISCUSSION TITLE
Proposal to simplify the terraform module inputs

### DISCUSSION THREAD

**Discussion by mjneuharth** [shibayan/terraform-azurerm-keyvault-acmebot#48] [Ideas]
I propose removing inputs for resources that are "internal" to the module.

Those configuration items are:
     
- app_service_plan_name
- storage_account_name
- app_insights_name
- workspace_name

As these are internal to the application, I think a better method would be to simply use a name base variable to compose these resource names.

For examples,

```bash
function_name_base = "acmebot-${random_string.random.result}"
```

And then internally we can compose the other resource names like this:

```bash
# This is untested but the idea is to take out characters that a storage account will not accept
local.storage_account_name = replace(lower(var.function_name_base), /[^a-z\\-0-9]/","")

app_service_plan_name = "plan-${function_name_base}"
storage_account_name = "st${local.storage_account_name}"
app_insights_name = "appi-${function_name_base}"
workspace_name = "log-${function_name_base}"
```

The storage account resource might be a little tricky but I think it could work.

**Comment by shibayan** [comment]
I think this is a good idea, and since we needed to take into account the differences in naming conventions for different Azure resources, we initially decided to leave all names up to the user, but I can understand why this would be redundant.

**Comment by mjneuharth** [comment]
Looks like you've implemented this. Thanks!

**Comment by cpressland** [comment]
@shibayan bringing this in will require us to recreate many resources, most worryingly the Storage Account. As I'm not super familiar with the upstream application, can it survive this loss of data or will I need to backup this data before proceeding?
